File: fizzBuzz.py
def fizzBuzz(lLow=0, lUpp=100, f=lambda n: print(n)):
    r = []
    nFizz, nBuzz = 3,5
    strFizz, strBuzz = "fizz", "buzz"

    for i in range(lLow, lUpp):
        isFizz = i % nFizz == 0
        isBuzz = i % nBuzz == 0
        if isFizz:
            r.append(strFizz)
        elif isBuzz:
            r.append(strBuzz)
        elif isFizz and isBuzz:
            r.append(strFizz+strBuzz)
        else:
            r.append(i)

        if callable(f):
            f(r[i])
    return r

r = fizzBuzz(lUpp=101)

print(r)

File: fizzBuzz.js
function fizzBuzz(lLow=0, lUpp=100, f=(n)=>console.log(n))
{
    let r = []
    let [nFizz , nBuzz] = [3,5];
    let [strFizz, strBuzz] = ["fizz", "buzz"]

    for(let i = lLow; i < lUpp; i++){
        let isFizz = i % nFizz == 0
        let isBuzz = i % nBuzz == 0
        if(isFizz && isBuzz){
            r.push(strFizz+strBuzz)
        }else if(isFizz){
            r.push(strFizz)
        }else if(isBuzz){
            r.push(strBuzz)
        }else{
            r.push(i)
        }
        
        if(typeof f == "function"){
            f(r[i])
        }
    }
    return r
}

let r = fizzBuzz(0, 101)
console.log("r:", r)

// #TODO USE GENERATORS

//# DENTRO DEL for, operaciones, book, go, ts, js, python, c++